system/lib/wasmfs/file.h

```diff
@@ -266,33 +293,47 @@ class DataFile::Handle : public File::Handle {
 
 class Directory::Handle : public File::Handle {
   std::shared_ptr<Directory> getDir() { return file->cast<Directory>(); }
+  void finishInsertingChild(const std::string& name,
+                            std::shared_ptr<File> child);
 
 public:
   Handle(std::shared_ptr<File> directory) : File::Handle(directory) {}
   Handle(std::shared_ptr<File> directory, std::defer_lock_t)
     : File::Handle(directory, std::defer_lock) {}
 
-  std::shared_ptr<File> getChild(const std::string& name) {
-    // Unlinked directories must be empty, without even "." or ".."
-    if (!getParent()) {
-      return nullptr;
-    }
-    if (name == ".") {
-      return file;
-    }
-    if (name == "..") {
-      return getParent();
-    }
-    return getDir()->getChild(name);
-  }
+  std::shared_ptr<File> getChild(const std::string& name);
```

Comment:
Should we add a comment here saying this will return the child if it is in the dcache, and otherwise will defer to the backend, which may create a new File on the fly and in that case that file would be returned?